MMEX/App/MMEXApp.swift

```diff
@@ -18,17 +18,18 @@ let log = Logger(
 struct MMEXApp: App {
     @StateObject private var env = EnvironmentManager(withStoredDatabase: ())
 
-    @AppStorage("isTrackingEnabled") private var isTrackingEnabled: Bool = true // Default is tracking enabled
-    @AppStorage("userID") private var userID: String = "" // Store user ID in AppStorage
-
     init() {
-        if userID.isEmpty {
-            userID = String(format: "ios_%@", TimestampString(Date()).string)
+        var userId = env.track.userId
+        if userId.isEmpty {
+            userId = String(format: "ios_%@", TimestampString(Date()).string)
```

Comment:
we should use a reference to update `env.track.userId`. i just fixed this via #262